core/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -845,6 +846,35 @@ public int getNodeLockId() {
         return shardIds;
     }
 
+    /**
+     * Find all the shards for this index, returning a map of the {@code NodePath} to the number of shards on that path
+     * @param index the index by which to filter shards
+     * @return a map of NodePath to count of the shards for the index on that path
+     * @throws IOException if an IOException occurs
+     */
+    public ObjectLongHashMap<NodePath> shardCountPerPath(final Index index) throws IOException {
```

Comment:
I am not a huge fan of returning a 3rd party lib class from a public interface. Can we rather return a `long[]` that index corresponds to the `NodePath[]` index in this class?